File: 118-pascals-triangle/118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> out = new LinkedList<>();
        for(int row = 0; row < numRows; row++){
            out.add(new LinkedList<Integer>());
            for(int col = 0; col <= row; col++){
                if(col == 0 || col == row){
                    out.get(row).add(1);        
                }else{
                    var sum = out.get(row-1).get(col-1) + out.get(row-1).get(col);
                    out.get(row).add(sum);
                }
            }
        }
        return out;
    }
}

File: 118-pascals-triangle/118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> dp;
        for(int row = 0; row < numRows; row++){
            vector<int> level;
            dp.push_back(level);
            for(int col = 0; col <= row; col++){
                if(col == 0 or col == row){
                    dp[row].push_back(1);
                }else{
                    int sum = dp[row-1][col-1] + dp[row-1][col];
                    dp[row].push_back(sum);
                }
            }
        }
        return dp;
    }
};